tests/controller_tests/test_filters.py

```diff
@@ -112,6 +189,154 @@ def test_get_outcome_categories(seeded_test_client):
     assert all(isinstance(cat, str) for cat in response_json)
 
 
+def populate_valid_test_data(test_client):
+    sub_1 = Submission(
+        submission_id="S-R03-3",
+        submission_date=datetime(2023, 5, 1),
+        reporting_period_start=datetime(2023, 4, 1),
+        reporting_period_end=datetime(2023, 4, 30),
+        reporting_round=3,
+        submission_filename="fake_name_drop_me",
+    )
+    sub_2 = Submission(
+        submission_id="S-R03-4",
+        submission_date=datetime(2023, 5, 1),
+        reporting_period_start=datetime(2023, 4, 1),
+        reporting_period_end=datetime(2023, 4, 30),
+        reporting_round=3,
+    )
+    sub_3 = Submission(
+        submission_id="S-R01-99",
+        submission_date=datetime(2023, 5, 1),
+        reporting_period_start=datetime(2023, 4, 1),
+        reporting_period_end=datetime(2023, 4, 30),
+        reporting_round=1,
+    )
+    db.session.add_all((sub_1, sub_2, sub_3))
+    db.session.commit()
+
+    read_sub = Submission.query.first()
+    read_sub_latest = Submission.query.filter(Submission.submission_id == "S-R03-4").first()  # The latest submission
+    read_sub_old = Submission.query.filter(
+        Submission.reporting_round == 1
+    ).first()  # replicate a submission from a previous round
+
+    organisation = Organisation(
+        organisation_name="Some new Org",
+        geography="Mars",
+    )
+    db.session.add(organisation)
+    db.session.commit()
+
+    read_org = Organisation.query.first()
+
+    prog1 = Programme(
+        programme_id="FHSF001",  # matches id for incoming ingest. Should be replaced along with all children
```

Comment:
You appear to have copied and pasted all these test setup steps from another test, including comments and names that refer specifically to things not happening in this test. You could either:

- refactor this test to remove all the misleading / unnecessary commenting / naming.
- or, if all the code is similar enough, re-use existing code rather than having a duplicate here

Reply:
I think I'd prefer option 1 as much of this data is non-pertinent to this test